src/com/carolinarollergirls/scoreboard/core/ScoreBoard.java

```diff
@@ -109,6 +110,7 @@
     public XmlScoreBoard getXmlScoreBoard();
 
     public enum Value implements PermanentProperty {
+        VERSION(String.class, ScoreBoardManager.getVersion()),
```

Comment:
There's more to the version than just that, I'd dump all the properties.

It'd also be better to pass these in when instantiating the scoreboard.

Reply:
> There's more to the version than just that, I'd dump all the properties.

Commit and timestamp are already in there. What use do you expect from build username and host?

> It'd also be better to pass these in when instantiating the scoreboard.

Why would that be better? It'd be more code, use more memory, and it'd diverge from every other instance where this value is used.

Reply:
> Commit and timestamp are already in there

Only if it's a custom build. Better to expose it all, just in case.

> Why would that be better?

It avoids a layering violation, the ScoreBoard shouldn't know that ScoreBoardManager exists.

Reply:
> Only if it's a custom build. Better to expose it all, just in case.

If we can't trust our release builds to be unambiguously numbered, we have a different level of problem.

> It avoids a layering violation, the ScoreBoard shouldn't know that ScoreBoardManager exists.

a) The ScoreBoard already does know ScoreBoardManager since you introduced it when moving Media over.
b) We have already discussed that the code is a mess in terms of layering and fixing that would amount to more or less a rewrite from the gorund up. In the meantime this one line doesn't add much damage but will make Smasher's life easier with the coversion tool.

Reply:
> If we can't trust our release builds to be unambiguously numbered, we have a different level of problem.

I'm not worried about our release builds, I'm worried about random builds.

> The ScoreBoard already does know ScoreBoardManager since you introduced it when moving Media over.

That should be fixed then.

> We have already discussed that the code is a mess in terms of layering and fixing that would amount to more or less a rewrite from the gorund up.

It's a lot better now that XML is gone.

Reply:
>I'm not worried about our release builds, I'm worried about random builds.

But those will have the git revision in there unless someone intentionally fiddles with the build process. And we can't protect against that.

Regarding the layering violation, I have moved the Version handling to utils, which I understand to be accessible from everywhere.